File: backend/src/main/java/com/military/doc/modules/project/entity/ProjectMasterData.java
package com.military.doc.modules.project.entity;

import com.baomidou.mybatisplus.annotation.*;
import com.military.doc.common.mybatis.JsonbTypeHandler;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName(value = "project_master_data", autoResultMap = true)
public class ProjectMasterData {
    @TableId(type = IdType.AUTO)
    private Long id;

    private Long projectId;

    @TableField(typeHandler = JsonbTypeHandler.class)
    private String equipmentInfo;

    @TableField(typeHandler = JsonbTypeHandler.class)
    private String tacticalIndicators;

    @TableField(typeHandler = JsonbTypeHandler.class)
    private String productTree;

    @TableField(typeHandler = JsonbTypeHandler.class)
    private String teamMembers;

    @TableField(typeHandler = JsonbTypeHandler.class)
    private String milestones;

    @TableField(typeHandler = JsonbTypeHandler.class)
    private String extendedFields;

    private Integer versionNo;
    private String status;

    @TableField(fill = FieldFill.INSERT)
    private Long createdBy;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private Long updatedBy;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;

    @TableLogic
    private Integer deleted;
}

File: frontend/src/api/project-master-data.ts
import api from './index'

export interface ProjectMasterData {
  id?: number
  projectId: number
  equipmentInfo?: string
  tacticalIndicators?: string
  productTree?: string
  teamMembers?: string
  milestones?: string
  extendedFields?: string
  versionNo?: number
  status?: string
  createdBy?: number
  createdAt?: string
  updatedBy?: number
  updatedAt?: string
}

export function getMasterData(projectId: number) {
  return api.get(`/projects/${projectId}/master-data`)
}

export function getFlattenedMasterData(projectId: number) {
  return api.get(`/projects/${projectId}/master-data/flattened`)
}

export function saveMasterData(projectId: number, data: Record<string, any>, operatorId: number) {
  return api.post(`/projects/${projectId}/master-data?operatorId=${operatorId}`, data)
}

export function getMasterDataSchema() {
  return api.get('/projects/1/master-data/schema')
}

export function getDefaultMasterData() {
  return api.get('/projects/1/master-data/default')
}

export function getNavigatorData(projectId: number) {
  return api.get(`/projects/${projectId}/navigator`)
}

export function extractMasterData(projectId: number) {
  return api.post(`/projects/${projectId}/master-data/extract`)
}
